Plugins/FreeRDP/Client/ConnectFreeRDP.cpp

```diff
@@ -1720,45 +1724,81 @@ void CConnectFreeRDP::mouseMoveEvent(QMouseEvent *event)
     if(!m_pContext) return;
     if(m_pParameter && m_pParameter->GetOnlyView()) return;
     UINT16 flags = PTR_FLAGS_MOVE;
-    SendMouseEvent(flags, event->pos());
+    SendMouseEvent(flags, pos,false);
 }
 
-void CConnectFreeRDP::mousePressEvent(QMouseEvent *event)
+void CConnectFreeRDP::slotMousePressEvent(Qt::MouseButtons buttons, QPoint pos)
 {
-    //qDebug(log) << __FUNCTION__ << event << event->buttons() << event->button();
-    if(!m_pContext) return;
-    if(m_pParameter && m_pParameter->GetOnlyView()) return;
-    UINT16 flags = PTR_FLAGS_DOWN;
-    if(event->button() & Qt::MouseButton::LeftButton)
+    // qDebug(FreeRDPConnect) << "CConnectFreeRDP::slotMousePressEvent" << buttons << pos;
+    if (!m_pContext) return;
+    if (m_pParameter && m_pParameter->GetOnlyView()) return;
+
+    UINT16 flags = 0;
+    bool isExtended = false;
+
+    if (buttons & Qt::MouseButton::LeftButton)
+    {
+        flags = PTR_FLAGS_DOWN | PTR_FLAGS_BUTTON1;
+    }
+    else if (buttons & Qt::MouseButton::RightButton)
+    {
+        flags = PTR_FLAGS_DOWN | PTR_FLAGS_BUTTON2;
+    }
+    else if (buttons & Qt::MouseButton::MiddleButton)
+    {
+        flags = PTR_FLAGS_DOWN | PTR_FLAGS_BUTTON3;
+    }
+    else if (buttons & Qt::MouseButton::ForwardButton)
+    {
+        flags = PTR_XFLAGS_DOWN | PTR_XFLAGS_BUTTON2;
+        isExtended = true;
+    }
+    else if (buttons & Qt::MouseButton::BackButton)
     {
-        flags |= PTR_FLAGS_BUTTON1;
-    } else if(event->button() & Qt::MouseButton::RightButton)
-    {   
-        flags |= PTR_FLAGS_BUTTON2;
-    } else if(event->button() & Qt::MouseButton::MiddleButton)
+        flags = PTR_XFLAGS_DOWN | PTR_XFLAGS_BUTTON1;
+        isExtended = true;
+    }
+    if (flags != 0)
     {
-        flags |= PTR_FLAGS_BUTTON3;
+        SendMouseEvent(flags, pos, isExtended);
     }
-    SendMouseEvent(flags, event->pos());
 }
 
-void CConnectFreeRDP::mouseReleaseEvent(QMouseEvent *event)
+void CConnectFreeRDP::slotMouseReleaseEvent(Qt::MouseButton button, QPoint pos)
```

Comment:
Please use:
    virtual void mouseReleaseEvent(QMouseEvent *event)

Reply:
My error, the code I used for debugging is old